### ISSUE TITLE
Literals in function parameters not disallowed

### ISSUE THREAD

**Issue by hellerve** [carp-lang/Carp#384]
Consider this expression:

```clojure
(defn x ["lol"] 1)
```

It is obviously wrong, since a string literal is used where a parameter name should be used. Carp doesn’t complain, though.

```clojure
(defn main [] (println* (x "a")))
```

So we call the function inside of another. Still, Carp doesn’t complain. Once we try to compile it, though, we will get the following error:

```clojure
carp: Invalid parameter: Str "lol"
CallStack (from HasCallStack):
  error, called at src/Emit.hs:703:28 in CarpHask-0.2.0.0-2URFTvpyTgUKakDgRrmlBo:Emit
```

This is too late in the pipeline. Ideally, we’d catch this as early as possible, i.e. when we define the function.

Cheers

P.S.: I’m happy to work on this, so you can assign it to me if you want to!

**Comment by hellerve** [comment]
This should be fixed with my overhaul of the errors in #396!

**Comment by eriksvedang** [comment]
yay!
